component/src/main/ballerina/ballerina/net/soap/natives.bal

```diff
@@ -81,80 +81,105 @@ public struct SoapError {
 }
 
 @Description { value:"SOAP client connector." }
-public connector SoapClient () {
+@Param { value:"serviceUri: Url of the service" }
+@Param { value:"connectorOptions: connector options" }
+public connector SoapClient (string serviceUri, http:Options connectorOptions) {
 
-    @Description { value:"Send Robust requests." }
+    @Description { value:"Send Robust requests.Sends the request and possibly receives an error"}
+    @Param { value:"path: Resource path " }
     @Param { value:"request: Request to be sent" }
-    @Param { value:"endpointURL: Endpoint the request should be sent" }
     @Return { value:"SoapError: The error if an error occurred" }
-    action sendRobust (Request request, string endpointURL) (SoapError) {
-        return null;
+    action sendRobust (string path, Request request) (SoapError) {
+        endpoint<http:HttpClient> httpConnector {
```

Comment:
Can't we move the endpoint and HttpClient creation to the SoapClient Connector global scope, without creating it in each action?